Src/WindowsServer/WindowsServer.Shared/Implementation/AzureRoleEnvironmentContextReader.cs

```diff
@@ -62,25 +65,97 @@ internal set
+
+                // Populating remote worker fields is done only in unit tests as it is 
+                // an expensive operation to communicate to remote object.
+                // The remote worker by default loads Azure ServiceRuntime assembly.
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.AssemblyName) == false)
+                {
+                    remoteWorker.AssemblyName = AzureRoleEnvironmentContextReader.AssemblyName;
+                }
+
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.Culture) == false)
+                {
+                    remoteWorker.Culture = AzureRoleEnvironmentContextReader.Culture;
+                }
+
+                if (string.IsNullOrEmpty(AzureRoleEnvironmentContextReader.PublicKeyToken) == false)
+                {
+                    remoteWorker.PublicKeyToken = AzureRoleEnvironmentContextReader.PublicKeyToken;
+                }
+
+                if (AzureRoleEnvironmentContextReader.VersionsToAttempt != null && AzureRoleEnvironmentContextReader.VersionsToAttempt.Length != 0)
+                {                    
+                    remoteWorker.VersionsToAttempt = AzureRoleEnvironmentContextReader.VersionsToAttempt;
+                }
 
-            if (roleEnvironment.IsAvailable == true)
+                bool success = remoteWorker.ReadAndPopulateContextInformation(ref this.roleName, ref this.roleInstanceName);
+                if (success)
+                {
+                    WindowsServerEventSource.Log.TroubleshootingMessageEvent("Successfully loaded assembly from remote worker in separate AppDomain");
+                }
+                else
+                {
+                    WindowsServerEventSource.Log.TroubleshootingMessageEvent("Failed loading assembly from remote worker in separate AppDomain. Application is assumed not be running in Azure Cloud service.");
+                }
+            }
+            catch (Exception ex)
             {
-                RoleInstance roleInstance = roleEnvironment.CurrentRoleInstance;
-                if (roleInstance != null)
+                WindowsServerEventSource.Log.TroubleshootingMessageEvent("AzureRoleEnvironmentContextReader Initialize failed : " + ex.ToString());
+            }
+            finally
+            {                
+                try
                 {
-                    this.roleInstanceName = roleEnvironment.CurrentRoleInstance.Id;
-                    Role role = roleInstance.Role;
-                    if (role != null)
+                    if (tempDomainToLoadAssembly != null)
                     {
-                        this.roleName = role.Name;
-                    }
+                        AppDomain.Unload(tempDomainToLoadAssembly);                                                
+                        WindowsServerEventSource.Log.TroubleshootingMessageEvent(tempDomainName + " AppDomain  Unloaded.");
```

Comment:
It is a bad pattern using the EventSource. Single method and string formatting inside is bad practice. Please split into mmultiple methods. Or at least have a `TempraryAppDomainTroubleshooint` method that accepts the domain name as an argument